src/django_upgrade/fixers/http_headers.py

```diff
@@ -0,0 +1,185 @@
+        kwarg_name_start_idx = find(tokens, arg_start_idx, name=NAME, src=kwarg_name)
+        if kwarg_name_start_idx > arg_end_idx:
+            continue
+
+        value_start_idx = find(tokens, arg_start_idx, name=OP, src="=")
+        erased_tokens = tokens[value_start_idx + 1 : arg_end_idx]
+        ends_with_comma = (
+            tokens[arg_end_idx].name == OP and tokens[arg_end_idx].src == ","
+        )
+        if ends_with_comma:
+            del tokens[arg_start_idx : arg_end_idx + 1]
+        else:
+            del tokens[arg_start_idx:arg_end_idx]
+        return erased_tokens
+    return []
+
+
+def insert_into_existing_headers_kwarg(
+    tokens: list[Token], i: int, *, kwarg_dict: dict[str, list[Token]]
+) -> None:
+    open_idx = find(tokens, i, name=OP, src="(")
+
+    headers_arg_start_idx = find(tokens, open_idx, name=NAME, src=HEADERS_KWARG)
+    headers_closing_idx = find(tokens, headers_arg_start_idx, name=OP, src="}")
+
+    new_tokens = []
+    for key, value in kwarg_dict.items():
+        new_tokens += [
+            Token(name="OP", src=","),
+            Token(name="UNIMPORTANT_WS", src=" "),
+        ]
+        new_tokens += [
+            Token(name="STRING", src=f'"{key}"'),
+            Token(name="OP", src=":"),
+            Token(name="UNIMPORTANT_WS", src=" "),
+        ]
+        new_tokens += value
+    tokens[headers_closing_idx:headers_closing_idx] = new_tokens
+
+
+def insert_headers_kwarg(
+    tokens: list[Token], i: int, *, kwarg_dict: dict[str, list[Token]]
+) -> None:
+    open_idx = find(tokens, i, name=OP, src="(")
+    func_args, close_idx = parse_call_args(tokens, open_idx)
+
+    new_tokens = []
+    if len(func_args) != 0:
+        if tokens[close_idx - 2].name != OP or tokens[close_idx - 2].src != ",":
+            new_tokens += [
+                Token(name="OP", src=","),
+            ]
+        new_tokens += [
+            Token(name="UNIMPORTANT_WS", src=" "),
+        ]
+
+    new_tokens += [
+        Token(name=NAME, src=HEADERS_KWARG),
+        Token(name="OP", src="="),
+        Token(name="OP", src="{"),
+    ]
+
+    for i, (key, value) in enumerate(kwarg_dict.items(), 1):
+        new_tokens += [
+            Token(name="STRING", src=f'"{key}"'),
+            Token(name="OP", src=":"),
+            Token(name="UNIMPORTANT_WS", src=" "),
+        ]
+        new_tokens += value
+        if i < len(kwarg_dict):
+            new_tokens += [
+                Token(name="OP", src=","),
+                Token(name="UNIMPORTANT_WS", src=" "),
+            ]
+
+    new_tokens += [
+        Token(name="OP", src="}"),
+    ]
+
+    tokens[close_idx - 1 : close_idx - 1] = new_tokens
+
+
+@fixer.register(ast.Call)
+def visit_Call(
+    state: State,
+    node: ast.Call,
+    parents: list[ast.AST],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        isinstance(node.func, ast.Name)
+        and "RequestFactory" in state.from_imports["django.test"]
```

Comment:
`AsyncRequestFactory` and `AsyncClient` are not handled at the moment, as the keyword arguments don't start with `HTTP_` I guess they would be harder to catch 🤔